app/assets/javascripts/posts.coffee

```diff
@@ -0,0 +1,36 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
+
+# Get the relevant posts and insert them in the "relevant to you" section
+Spaces.updateRelevantPosts = () ->
+	$.ajax
+		url: 'posts/relevant'
+		type: 'GET'
+		success: (data, textStatus, jqXHR) ->
+			$(".relevant_posts .posts").html data
+		error: (jqXHR, textStatus, errorThrown) ->
+			console.log "Error updating posts: "+errorThrown
+
+# After adding a post:
+#	- Clear the form
+#	- Add the post to 'most recent' section
+#	- Update both forms to include any new tags
+Spaces.addPostCallback = (data) ->
+	($ '.modal').modal 'hide'
+	($ '.all_posts .posts').prepend data
+	($ "#post_short")[0].value = ""
+	($ "#post_long")[0].value = ""
+	($ "#post_tag_list")[0].selectize.clear()
+	Spaces.updatePostForm()
+	Spaces.updateUserForm()
```

Comment:
this works in this very small example, but if you try doing things like this as your app grows you end up with spaghetti.

You are really doing three things here - a better architecture is to emit a DOM event in this callback, like `$('body').trigger 'postAdded'`, and then, in different controllers, listen for the emitted event on the DOM and add the post + update forms.